extensions/amp-live-list/0.1/amp-live-list.js

```diff
@@ -379,13 +404,42 @@ export class AmpLiveList extends AMP.BaseElement {
    * @private
    */
   sortByDataSortTime_(a, b) {
-    const aTime = Number(a.getAttribute('data-sort-time'));
-    const bTime = Number(b.getAttribute('data-sort-time'));
-    user.assert(aTime > 0, '`data-sort-time` attribute must exist and value' +
-        ' must be a number. Found %s.', aTime);
-    user.assert(bTime > 0, '`data-sort-time` attribute must exist and value' +
-        ' must be a number. Found %s.', aTime);
-    return aTime - bTime;
+    return this.getSortTime_(a) - this.getSortTime_(b);
+  }
+
+  /**
+   * @param {!Element} elem
+   * @return {number}
+   * @private
+   */
+  getSortTime_(elem) {
+    return this.getTimeAttr_(elem, 'data-sort-time');
+  }
+
+  /**
+   * @param {!Element} elem
+   * @return {number}
+   * @private
+   */
+  getUpdateTime_(elem) {
+    // Since update-time is optional just return 0 if it doesnt exist.
+    if (!elem.hasAttribute('data-update-time')) {
+      return 0;
+    }
+    return this.getTimeAttr_(elem, 'data-update-time');
+  }
+
+  /**
+   * @param {!Element} elem
+   * @return {number}
+   * @private
+   */
+  getTimeAttr_(elem, attr) {
+    const time = Number(elem.getAttribute(attr));
```

Comment:
will add TODO for now. sort time is definitely easy to cache since we treat it as immutable (even if the publisher mistakenly updates it). update time we have to read the property and convert anyways to do the comparison.